.github/workflows/integration_test.yml

```diff
@@ -0,0 +1,42 @@
+# TODO: Support running all integration tests for master branch when
+# it's possible to run self-hosted runners in parallel.
+name: Integration Test
+
+on: 
+#  push:
+#    branches:
+#      - "master"
```

Comment:
Why did you comment out? Please convert this PR to draft if the change is not ready for review yet. Also you don't need the quotation marks for the branch name as it doesn't contain whitespace characters.

Reply:
The comment out part was meant to serve in the next PR when the server supports multiple self-hosted runners, I left a TODO comment above but I can just remove it and add it when it becomes relevant. I'll remove the quotation marks in the next commit.